app/uk/gov/hmrc/securemessage/models/core/CustomerEnrolment.scala

```diff
@@ -30,8 +30,11 @@ object CustomerEnrolment {
     Json.reads[CustomerEnrolment]
   }
 
-  def parse(enrolmentString: String): CustomerEnrolment = {
+  def parse(enrolmentString: String): Either[String, CustomerEnrolment] = {
     val enrolment = enrolmentString.split('~')
-    CustomerEnrolment(enrolment.head, enrolment(1), enrolment.last)
+    enrolment.size match {
+      case 1 => Left("Unable to bind a CustomerEnrolment")
+      case _ => Right(CustomerEnrolment(enrolment.head, enrolment(1), enrolment.last))
+    }
```

Comment:
wouldn't you want to raise the error if the enrolment.size is anything but 2?

Reply:
I will change it.